src/main/java/com/example/demo/movies/MovieService.java

```diff
@@ -0,0 +1,83 @@
+package com.example.demo.movies;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+
+@Service
+@RequiredArgsConstructor
+public class MovieService {
+
+    private final MovieRepository movieRepository;
+
+    public List<Movie> getMovies() {
+        return movieRepository.findAll();
+    }
+
+    public void addNewMovie(Movie movie) {
+        Optional<Movie> movieOptional = movieRepository
+                .findMovieByTitle(movie.getTitle());
+
+        if (movieOptional.isPresent()) {
+            throw new IllegalStateException("This movie already exists");
+        }
+
+        movieRepository.save(movie);
+    }
+
+    public void deleteMovie(Integer movieId) {
+
+        boolean exists = movieRepository.existsById(movieId);
+
+        if (!exists) {
+            throw new IllegalStateException("Movie with id " + movieId + " does not exist");
+        }
+
+        movieRepository.deleteById(movieId);
+    }
+
+    @Transactional
+    public void updateMovie(Integer movieId, String title, Integer categoryId, String imdbId, Double imdbScore, String description, LocalDate releaseDate) {
+
+        Movie movie = movieRepository.findById(movieId)
+                .orElseThrow(() -> new IllegalStateException("Movie with id " + movieId + " does not exist"));
+
+        if ( title != null &&
```

Comment:
Nothing to modify here at the moment, but once we have a DTO for Movie entity, we can make these checks in the MovieDto class itself